src/StrawberryShake/Client/src/Transport.WebSockets/ISocketOperationManager.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using StrawberryShake.Transport.WebSockets.Messages;
+
+namespace StrawberryShake.Transport.Subscriptions
+{
+    public interface ISocketOperationManager
```

Comment:
The socket operation manager manages the lifecycle of different operations over the socket.
It starts operation by calling the protocol and subscribes to it's updates